Dockerfile

```diff
@@ -1,15 +1,24 @@
+# syntax=docker/dockerfile:1
+FROM node:16-bullseye as build
+RUN apt-get update && apt-get install -y openssl
+WORKDIR /app
+COPY . .
+RUN yarn install && yarn build
+
 FROM node:16-bullseye as base
 RUN apt-get update && apt-get install -y openssl
 WORKDIR /app
 ENV NODE_ENV=production
-ADD yarn.lock package.json ./
-RUN npm rebuild bcrypt --build-from-source
-RUN yarn install --production
+COPY --from=build /app/build ./
```

Comment:
Why are there all of these Dockerfile changes in here?

These are unrelated to TypeScript. Please keep this PR scoped down as it will be a lot easier to review and merge in

Reply:
Unless I'm missing something; can't you just add `&& yarn build` here?

Reply:
And make all these other changes in another PR